presto-spi/src/main/java/com/facebook/presto/spi/security/ConnectorIdentity.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.security;
+
+import java.security.Principal;
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonMap;
+import static java.util.Objects.requireNonNull;
+
+public class ConnectorIdentity
+{
+    private final String user;
+    private final Optional<Principal> principal;
+    private final Optional<SelectedRole> role;
+
+    public ConnectorIdentity(String user, Optional<Principal> principal, Optional<SelectedRole> role)
+    {
+        this.user = requireNonNull(user, "user is null");
+        this.principal = requireNonNull(principal, "principal is null");
+        this.role = requireNonNull(role, "role is null");
+    }
+
+    public String getUser()
+    {
+        return user;
+    }
+
+    public Optional<Principal> getPrincipal()
+    {
+        return principal;
+    }
+
+    public Optional<SelectedRole> getRole()
+    {
+        return role;
+    }
+
+    public Identity toIdentity()
```

Comment:
I agree. This method is used in single place only. Gonna do the same but inline in that place.